pygmt/datasets/earth_night.py

```diff
@@ -0,0 +1,97 @@
+"""
+
+from collections.abc import Sequence
+from typing import Literal
+
+import xarray as xr
+from pygmt.datasets.load_remote_dataset import _load_remote_dataset
+
+__doctest_skip__ = ["load_black_marble"]
+
+
+def load_black_marble(
+    resolution: Literal[
+        "01d",
+        "30m",
+        "20m",
+        "15m",
+        "10m",
+        "06m",
+        "05m",
+        "04m",
+        "03m",
+        "02m",
+        "01m",
+        "30s",
+    ] = "01d",
+    region: Sequence[float] | str | None = None,
+) -> xr.DataArray:
+    r"""
+    Load NASA Black Marble images in various resolutions.
+
+    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_earth_daynight.jpg
+       :width: 80%
+       :align: center
+
+       Earth day/night dataset.
+
+    The images are downloaded to a user data directory (usually
+    ``~/.gmt/server/earth/earth_night/``) the first time you invoke this function.
+    Afterwards, it will load the image from the data directory. So you'll need an
+    internet connection the first time around.
+
+    These images can also be accessed by passing in the file name
+    **@earth_night**\_\ *res* to any image processing function or plotting method. *res*
+    is the image resolution (see below).
+
+    Refer to :gmt-datasets:`earth-daynight.html` for more details about available
+    datasets, including version information and references.
+
+    Parameters
+    ----------
+    resolution
+        The image resolution. The suffix ``d``, ``m``, and ``s`` stand for arc-degree,
+        arc-minute, and arc-second.
+
+    region
+        The subregion of the image to load, in the form of a sequence [*xmin*, *xmax*,
+        *ymin*, *ymax*].
+
+    Returns
+    -------
+    image
+        The NASA Black Marble image. Coordinates are latitude and longitude in degrees.
+
+    Note
+    ----
+    The registration and coordinate system type of the returned
+    :class:`xarray.DataArray` image can be accessed via the GMT accessors (i.e.,
+    ``image.gmt.registration`` and ``image.gmt.gtype`` respectively). However, these
+    properties may be lost after specific image operations (such as slicing) and will
+    need to be manually set before passing the image to any PyGMT data processing or
+    plotting functions. Refer to :class:`pygmt.GMTDataArrayAccessor` for detailed
+    explanations and workarounds.
+
+    Examples
+    --------
+
+    >>> from pygmt.datasets import load_black_marble
+    >>> # load the default image (pixel-registered 1 arc-degree image)
+    >>> image = load_black_marble()
+    """
+    image = _load_remote_dataset(
+        name="earth_night",
+        prefix="earth_night",
+        resolution=resolution,
+        region=region,
+        registration="pixel",
+    )
+    # If rioxarray is installed, set the coordinate reference system
+    if hasattr(image, "rio"):
+        image = image.rio.write_crs(input_crs="OGC:CRS84")
```

Comment:
This line is not covered by a unit test. Should I add one, or we can ignore it?

Reply:
It should be covered when rioxarray is installed, right?

Reply:
Or do we have to import rioxarray to register the rio accessor

Reply:
The equivalent line in `load_blue_marble` is covered here in `test_grdimage_image.py` which has an `import rioxarray` at the top of the file:

https://github.com/GenericMappingTools/pygmt/blob/66f258aa8bb98d55b6768067ad00b60cd06c89b4/pygmt/tests/test_grdimage_image.py#L18

We don't test `load_black_marble` anywhere else except in `test_datasets_earth_night.py`, and since `rioxarray` is not imported in that file, this line isn't covered because there is no `rio` accessor.

Reply:
Maybe we can cover this line in https://github.com/GenericMappingTools/pygmt/pull/3115/files#r1780370539?

Reply:
> since `rioxarray` is not imported in that file, this line isn't covered because there is no `rio` accessor.

This is not ideal. So even if rioxarray is installed, the `rio` accessor may still not exist because users don't import the rioxarray package.

I think we should add the following lines to this file (and any other files that try to access the `rio` accessor):

https://github.com/GenericMappingTools/pygmt/blob/66f258aa8bb98d55b6768067ad00b60cd06c89b4/pygmt/datasets/tile_map.py#L21-L23

Reply:
xref: https://github.com/GenericMappingTools/pygmt/pull/3323

Reply:
Ok, done at commit 66183ed40fae857ca0f29b19dd69d1cf87bd5811. I've also applied this to `pygmt/datasets/earth_day.py`, hopefully you don't mind.